text/0000-CNB-user-research-2021.md

```diff
@@ -0,0 +1,85 @@
+# Meta
+[meta]: #meta
+- Name: CNB User Research 2021
+- Start Date: 2020-12-10
+- Author(s): sampeinado, speinado@vmware.com
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Background
+The purpose of this study is to understand the major pain points buildpack authors and app developers experience in using Cloud Native Buildpacks and how these issues slow the adoption of CNBs as an emerging standard in containerization. VMware has donated funding for the recruitment and compensation of 15-20 interview subjects to help the CNB project grow. 
+ 
+Given our contributors’ strong understanding of buildpacks and their spec, the project is at risk of a blindspot towards novice buildpack authors who want to create value quickly. The CNB spec enables more modularity than the v2 spec, but its API is also significantly more complex. For our ecosystem of buildpacks to grow, and therefore the number of app types and use cases buildpacks support to grow, we need to develop a better understanding of where buildpack authors struggle and how we can support them. 
+ 
+In addition, we want to better understand why app developers switch to CNBs and what challenges they face when doing so. As a project, we have clear assumptions about why CNBs are valuable relative to other containerization tools, but we don’t know which reasons are most important to which users. Without this clarity, our ability to market the project’s strengths and prioritize the most valuable features is limited. 
+
+# Objectives
+<ol>
+<li>Understand how buildpack authors currently go about writing buildpacks and customizing builds and what goals they are trying to accomplish by doing so</li>
+<ol>
+<li>Evaluate pain points buildpack authors are experiencing </li>
+<li>Evaluate the relative usability of proposals like <a href="https://github.com/sclevine/packfile">packfile</a> or <a href="https://github.com/kr/heroku-buildpack-inline">inline buildpacks</a> </li>
+</ol>
+<li>Define the different segments of app developers using buildpacks, their goals, challenges, and motivations to switch to buildpacks from other tools</li>
+<ol>
+<li>Identify a tipping point at which users can be considered “converted”</li>
+<li>Enable us to become more strategic in our communication in conference talks, blog posts, and working group </li>
+</ol>
+<li>Provide survey responses to key topics the project is currently working on</li>
+<ol>
+<li>How and where they find information today</li>
+<li>Which platforms they use CNBs on</li>
+<li>Expectations around inner loop development</li>
+</ol>
+<li>Anti-goals</li>
+<ol>
+<li>Who’s using what buildpacks brand</li>
+<li>Usability of the `pack` CLI, except as it relates to buildpack authorship</li>
+<li>Challenges of platform implementors integrating the lifecycle</li>
+<li>Challenges of platform operators doing governance</li> 
```

Comment:
Not sure I understand this one, but it's an anti-goal so maybe that doesn't matter 😆.

Reply:
the first is just to say, the point is not to gather competitive research on whether users are more likely to use Google's buildpacks or Paketo's buildpacks or whatever

Reply:
the third is to say that we aren't trying to learn about how to make it easier for someone building a platform to add buildpacks functionality to the platform, we are interested in end users

Reply:
the fourth is to say that we aren't trying to learn about the the challenges people experience when trying to standardize dependencies and OSes across their organization, which is one of the value props of buildpacks, but not in scope for this study